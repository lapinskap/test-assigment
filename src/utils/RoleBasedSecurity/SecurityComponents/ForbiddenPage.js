import PropTypes from 'prop-types';
import React from 'react';
import { Alert } from 'reactstrap';
import __ from '../../Translations';

const ForbiddenPage = ({ permission }) => (
  <Alert color="danger">
    {__('Nie masz uprawnień do oglądania tej strony. Brakujące pozwolenie: ({0})', [permission])}
  </Alert>
);

export default ForbiddenPage;

ForbiddenPage.propTypes = {
  permission: PropTypes.string,
};

ForbiddenPage.defaultProps = {
  permission: '',
};
