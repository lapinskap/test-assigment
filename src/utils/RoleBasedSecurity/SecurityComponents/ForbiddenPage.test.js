import React from 'react';
import renderer from 'react-test-renderer';
import ForbiddenPage from './ForbiddenPage';

describe('<ForbiddenPage />', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with default state', () => {
    const tree = renderer
      .create(<ForbiddenPage
        permission="random:random"
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
