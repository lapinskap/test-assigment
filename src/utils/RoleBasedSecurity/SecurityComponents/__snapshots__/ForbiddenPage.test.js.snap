// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForbiddenPage /> renders correctly with default state 1`] = `
<div
  className="alert alert-danger fade"
  role="alert"
>
  Nie masz uprawnień do oglądania tej strony. Brakujące pozwolenie: (random:random)
</div>
`;
